R/test_result.R

```diff
@@ -1,127 +1,143 @@
-#'   \item \code{num_total}: Number of tests
-#'   \item \code{errors}: Vector of errors found. (\code{NULL})
-#' }
-#' @param incorrect A character string to display if at least one test fails.
-#'   This character string will be run through \code{glue::\link[glue]{glue_data}} with
-#' \itemize{
-#'   \item \code{num_correct}: Number of correct tests
-#'   \item \code{num_total}: Number of tests
-#'   \item \code{errors}: Vector of errors found
-#' }
+#' @param ... \code{\link{pass_if}} or \code{\link{fail_if}} \code{\link{condition}}s to check
+#' @template correct
+#' @template incorrect
 #' @template grader_args
 #' @template learnr_args
-#' @param ... ignored
+#' @param glue_correct A glue string that returns the final correct message displayed.
+#'    Defaults to \code{getOption("gradethis_glue_correct_test")}, e.g.,
+#'    2/2 correct! Absolutely fabulous!.
+#' @param glue_incorrect A glue string that returns the final correct message displayed.
+#'    Defaults to \code{getOption("gradethis_glue_correct_test")}, e.g.,
+#'    1/2 correct! Try it again; next time's the charm!.
+#'
+#' @return a \code{\link{graded}} object whether or not all test cases passed.
+#'   If \code{\link{pass_if}} case is \code{TRUE} it is considered as passed.
+#'   If \code{\link{fail_if}} case is \code{FALSE} it is also considered as passed.
+#'   The message, by default, will report the number of passed conditions
+#'   over the total number of conditions
 #'
-#' @return a \code{grader_graded} structure from \code{\link{graded}} containing
-#'   a formatted \code{correct} or \code{incorrect} message.
-#' @seealso \code{test}
+#' @seealso \code{\link{check_code}}, \code{\link{check_result}}, and \code{\link{test_result}}
 #' @export
 #' @examples
 #' \dontrun{grading_demo()}
+#'
+#' example_function <- function(x){
+#'   return(x + 1)
+#' }
+#' test_result(
+#'   pass_if(~ .result(3) == 4),
+#'   pass_if(~ .result(10) == 11),
+#'   grader_args = list(),
+#'   learnr_args = list(last_value = example_function, envir_prep = new.env())
+#' )
+#'
+#' test_result(
+#'   pass_if(~ .result(3) == 4),
+#'   fail_if(~ .result(10) == 11),
+#'   grader_args = list(),
+#'   learnr_args = list(last_value = example_function, envir_prep = new.env())
+#' )
 test_result <- function(
   ...,
-  correct = "{num_correct}/{num_total} correct! {random_praise()}",
-  incorrect = "",
+  correct = NULL,
+  incorrect = NULL,
   grader_args = list(),
-  learnr_args = list()
+  learnr_args = list(),
+  glue_correct = getOption("gradethis_glue_correct_test"),
+  glue_incorrect = getOption("gradethis_glue_incorrect_test")
 ) {
-  tests <- grader_tests(...)
-  chkm8_class(tests, "grader_tests")
-  chkm8_single_character(correct)
-  chkm8_single_character(incorrect)
 
-  user_answer <- get_user_code(grader_args$user_quo)
+  conditions <- list(...)
+  chkm8_item_class(conditions, "grader_condition")
+
+  test_results <- purrr::map(conditions, pass_fail_condition_modify,
```

Comment:
Is this the only place we use `purrr`? While I like the tidyverse coding style, it is an unnecessary dependency.  (Use `lapply`)

Reply:
@garrettgman uses it in his `unpipe` and `order_calls` files. We could re-write those, the main one that isn't a straight `lapply` is `purrr::safely`

Reply:
ok.  keep the `purrr`. Thanks